### PR TITLE
introduce quarterly and yearly base frequency

### PR DESCRIPTION
**Pull request by vafl**
and use them for m3 and m4

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/time_feature/lag.py [hunk 1]
```diff
@@ -85,9 +85,14 @@ def _make_lags_for_month(multiple, num_cycles=3):
     # multiple, granularity = get_granularity(freq_str)
     offset = to_offset(freq_str)
 
-    if offset.name == "M":
+    if offset.name == "A-DEC":  # Y
+        lags = []
+    elif offset.name == "Q-DEC":  # Q
+        assert offset.n == 1
```
Comment (jaheba) [inline src/gluonts/time_feature/lag.py:91]: Can we add a message here?

File: src/gluonts/time_feature/lag.py [hunk 2]
```diff
@@ -85,9 +85,14 @@ def _make_lags_for_month(multiple, num_cycles=3):
     # multiple, granularity = get_granularity(freq_str)
     offset = to_offset(freq_str)
 
-    if offset.name == "M":
+    if offset.name == "A-DEC":  # Y
```
Comment (jaheba) [inline src/gluonts/time_feature/lag.py:88]: Can we have make the comment a bit more expressive? I assume these are aliases for pandas frequencies?

**Review by jaheba — COMMENTED** [review]
Should we also add the new frequencies to `get_seasonality` in `time_feature`?

File: src/gluonts/time_feature/lag.py (hunk 1 above)
Comment (vafl) [reply src/gluonts/time_feature/lag.py:91]: You mean a message saying that we don't use features for years?
I think that is just the correct behavior. There are no useful features for years I think

File: src/gluonts/time_feature/lag.py (hunk 2 above)
Comment (vafl) [reply src/gluonts/time_feature/lag.py:88]: Yeah its just the normalized frequency string (like we already use for W-SUN)

File: src/gluonts/time_feature/lag.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/time_feature/lag.py:91]: Sorry, I meant to the assert statement.

**Review by jaheba — APPROVED** [review]

**Pull request merged**